### DISCUSSION TITLE
When using o1 models, the reasoning token is not counted in the token usage

### DISCUSSION THREAD

**Discussion by khfung** [danny-avila/LibreChat#4201] [Troubleshooting]
### What happened?

When using o1 models, the reasoning token is not counted in the token usage. 
It only counted the completion and prompt but not the reasoning token usage.

### Steps to Reproduce

1. use the o1 models
2. see the mongo express database, and there are only completion and prompt but no reasoning.

### Screenshots

![image](https://github.com/user-attachments/assets/ec81c30c-1d03-44a1-971f-242ed0fef348)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
It is counted, I went to great lengths to assure this. Are you using the official API?

**Reply by khfung** [reply]
> It is counted, I went to great lengths to assure this. Are you using the official API?
Thanks
I am not using the official API because I have not reached tier 5; instead, I am using OpenRouter.
When I check openrouter, it shows like this
If I use Openrouter, it won't include the reasoning token on the Mongodb or token usage?
{
    "data": {
        "id": "gen-aSG7DDxNk84rYBSqmRKPI87Afm0S",
        "total_cost": 0.03880305,
        "upstream_id": "chatcmpl-AAKJOBp71d2BJGBgfVB3KikgZx1A3",
        "created_at": "2024-09-23T09:06:32.328985+00:00",
        "model": "openai/o1-preview",
        "app_id": 150064,
        "streamed": false,
        "cancelled": false,
        "provider_name": "OpenAI",
        "latency": 10106,
        "moderation_latency": 75,
        "generation_time": 0,
        "finish_reason": "stop",
        "tokens_prompt": 12,
        "tokens_completion": 198,
        "native_tokens_prompt": 13,
        "native_tokens_completion": 650,
        "native_tokens_reasoning": 448,
        "num_media_prompt": null,
        "num_media_completion": null,
        "origin": "https://librechat.ai/",
        "usage": 0.03880305
    }
}

**Reply by danny-avila** [reply]
Correct, because I'm only accounting for OpenAI's usage data structure. I can look into openrouter's but it's clearly different and the source of the issue.

**Reply by khfung** [reply]
Thanks again in advance.
Also, I saw someone on your Discord suggested setting an OpenRouter endpoint to an OpenAI reverse proxy. Is it advisable to do that? (I've set it up in the same way as well.)
I plan to use only OpenAI's model. Will the token usage count differ from the official one?
